### DISCUSSION TITLE
Show only the region on the track and hide the rest of the track

### DISCUSSION THREAD

**Discussion by logesh2496** [katspaugh/wavesurfer.js#3162] [Q&A]
Hi @katspaugh ,

I'm looking to hide the entire waveform of the track and only show the region portion of the waveform. Trying to recreate the below design from the DAW, but I only have one audio source for all the tracks. 
![image](https://github.com/katspaugh/wavesurfer.js/assets/55389513/f50d7d4e-fbc1-4d6d-ad86-30c23c1f9838)

I also tried creating multiple regions as shown below

Region 1: 0s-10s
Region 2: 10s-rest of the track with dark color so it's not visible.

But the problem with this is that I cannot resize the region: 1. Changing the start time of the Region_2 as the Region_1 resizes could also be a solution.

Thanks

**Comment by katspaugh** [comment]
Other than covering the invisible portion with an opaque region, you could play around with [CSS masking/clipping](https://css-tricks.com/clipping-masking-css/) or perhaps with the mix-blend-mode.

**Comment by logesh2496** [comment]
I have created opaque regions before and after my target regions, is there any event that triggers while resizing so that I can adjust the opaque regions on the fly?
When a target region extends, how do I change the start time of the opaque region during resizing?

<img width="1135" alt="image" src="https://github.com/katspaugh/wavesurfer.js/assets/55389513/c4256468-1829-4b21-b7ff-ddd017baf302">

**Reply by katspaugh** [reply]
There's a region-updated event.
See https://wavesurfer-js.org/docs/modules/plugins_regions

**Reply by logesh2496** [reply]
but the event only triggers when the update is finished. is there any alternate event that's called during the resize process?

**Reply by katspaugh** [reply]
Individual regions emit an `update` event during resizing. See https://wavesurfer-js.org/docs/types/plugins_regions.RegionEvents
Please refer to the docs or the source code in case of further questions, I think it's small and simple enough to follow.

**Reply by logesh2496** [reply]
oh yes, sorry about that. thanks
